src/LaunchDarkly.TestHelpers/JsonAssertions.cs

```diff
@@ -0,0 +1,371 @@
+    public struct JsonTestValue
+    {
+        internal readonly string Raw;
+        internal readonly JToken Parsed;
+
+        /// <summary>
+        /// True if there is a value (that is, the original string was not a null reference).
+        /// </summary>
+        public bool IsDefined => Parsed != null;
+
+        /// <summary>
+        /// Shortcut for creating an undefined value, equivalent to <c>JsonOf(null)</c>.
+        /// </summary>
+        public static JsonTestValue NoValue => JsonOf(null);
+
+        private JsonTestValue(string raw, JToken parsed)
+        {
+            Raw = raw;
+            Parsed = parsed;
+        }
+
+        /// <summary>
+        /// Creates a <c>JsonTestValue</c> from a string that should contain JSON.
+        /// </summary>
+        /// <remarks>
+        /// This method fails immediately for any string that is not well-formed JSON. However, if
+        /// it is a null reference, it returns an "undefined" instance that will return <c>false</c>
+        /// from <see cref="IsDefined"/>.
+        /// </remarks>
+        /// <param name="raw">the input string</param>
+        /// <returns>a <c>JsonTestValue</c></returns>
+        /// <exception cref="FormatException">for malformed JSON</exception>
+        public static JsonTestValue JsonOf(string raw)
+        {
+            if (raw is null)
+            {
+                return new JsonTestValue(null, null);
+            }
+            try
+            {
+                return new JsonTestValue(raw, JToken.Parse(raw));
+            }
+            catch (Exception e)
+            {
+                throw new FormatException("not valid JSON (" + e + "): " + raw);
+            }
+        }
+
+        internal static JsonTestValue OfParsed(JToken parsed)
+        {
+            return new JsonTestValue(parsed is null ? null : parsed.ToString(Formatting.None), parsed);
+        }
+
+        /// <summary>
+        /// Creates a <c>JsonTestValue</c> by serializing an arbitrary value to JSON.
+        /// </summary>
+        /// <remarks>
+        /// For instance, <c>JsonFromValue(true)</c> is equivalent to <c>JsonOf("true")</c>.
+        /// This only works for types that are supported by Newtonsoft.Json's default
+        /// reflection-based serialization mechanism.
+        /// </remarks>
+        /// <param name="value">an arbitrary value</param>
+        /// <returns>a <c>JsonTestValue</c></returns>
+        public static JsonTestValue JsonFromValue(object value) =>
+            OfParsed(JToken.FromObject(value));
+
+        /// <summary>
+        /// If this value is a JSON object, return the value of the specified property or
+        /// an undefined value if there is no such property.
+        /// </summary>
+        /// <param name="name">the property name</param>
+        /// <returns>the value, if any</returns>
+        /// <exception cref="InvalidOperationException">if the current value is not an object</exception>
```

Comment:
Is this `exception` tag incorrect? It doesn't look like `XunitException` is a subtype. Also applies to `RequiredProperty`.